packages/@ember/-internals/container/lib/container.ts

```diff
@@ -470,8 +468,12 @@ function buildInjections(
   typeInjections: Injection[],
   injections: Injection[]
 ): BuildInjectionsResult {
+  let injectionsHash = {};
+
+  injectionsHash[OWNER as any] = container.owner!;
```

Comment:
So, if we add LEGACY_OWNER here, it will result in an enumerable LEGACY_OWNER property on Ember.CoreObject _unless_ we delete it or purposefully ignore it, which seems bad.

Given that Symbol _works_ with `Object.assign`, I think it should be fine in most real use cases to just do this. The only one we've seen that relied on enumerability used `setOwner`, not the container API.

If we do think we need it though, I think we have to accept that CoreObject will need to do a bit more work to ignore it.

Reply:
Sadly, I do think we should do this. An idea that @pzuraq came up with in chat was to make this use `setOwner` (and therefore set `LEGACY_OWNER`) but also add a `set [LEGACY_OWNER]() {}` on `CoreObject` to "black hole" the property (and avoid enumerability).